features/atcryptoauth/README.md

```diff
@@ -0,0 +1,174 @@
+- Only ECDSA sign and verify operations are implemented.
+- The device is commissioned with a minimilistic configuration to allow ECDSA operations.
+- Locking configuration zone (EEPROM region containing config) is locked as it is necessary to use the device.
+- Locking data zone is not required for sample or demo purposes.
+- The device is not configured to authenticate driver commands.
+- Public keys are not stored in the device but generated whenever required.
+
+## ATCAECC508A configuration
+The device configuration should be **locked** before use. To achieve demo requirements it is commissioned with a minimilistic configuration. The commissioning can be done using the sample commissioning app contained in the driver code. This app commissions following configuration in the device:
+- Data slots 0-7 are configured to contain ECC Private keys.
+- Slot 8 is configured to store a x509 certificate.
+- Slot 9-14 is configured to store ECC Public keys.
+
+## Commissioning Application.
+Commissioning app is part of the driver source. It is gaurded under compile flag ```MBED_CONF_ATCAECC_APP_ENABLE``` and disabled by default in ```mbed_lib.json```. In order to do commissioning mbed-os should be built with this flag enabled.
+
+On start it prints following options:
+```
+Options:
+a) Commission ATCAECC508A.
+b) Dump config.
+c) Generate Private key.
+d) Run SSL client example.
+
+Enter choice:
+```
+Commissioning is done in two steps:
+- First, pressing character 'a' configures the device as described above. This step challenges the user to lock the config. On pressing 'y' the configuration is locked. Note: this step is **irreversible**.
+- Next, pressing 'c' generated a persistent Private key in slot 0. This key can be referred as *Key Id 0* with the driver API.
+
+***The demo config is done!***
+
+Note: The device allows re-generating Private keys. Any Public key or Certificates generated using same Key Id will be invalidated on re-generating the Private key.
+
+## Generating certificates
+A self signed certificate can be generated using [modified](https://github.com/ARMmbed/mbedtls/pull/1360) ```cert_write.exe``` application. Steps are as follows:
+- Connect ATCAECC508A shield on an Mbed platform.
+- Flash Mbed Platform with the sample commissioning application. Reset the board.
+- Run modified ```cert_write.exe``` with parameters shown below:
+
+```
+mbedtls/programs/x509/cert_write.exe subject_key=remote0COM18 issuer_key=remote0COM18 issuer_name=CN=Cert,O=mbed TLS,C=UK authority_identifier=0 output_file=cert.pem
+```
+Above command on success generates a certificate in ```cert.pem``` file. This certificate can be inspected with following command:
+```
+openssl x509 -in cert.pem -noout -text
+```
+
+Modified ```cert_write.exe``` takes parameters ```subject_key``` and ```issuer_key``` formatted in a special way to identify HW keys. 
+The format is ```remote0COM18```. Here:
```

Comment:
Please put a separator between the parts, e.g. `remote:0:COM18` or `///opaque_pk/ATCA/0:COM18`